src/signature_util.cc

```diff
@@ -46,73 +45,81 @@ template <size_t N> constexpr std::array<uint8_t, N> hex2pubkey(const char (&hex
   return pubkey;
 }
 
+std::optional<std::array<uint8_t, 32>> getEd25519PubKey(std::string pubkey) {
+#ifdef PROXY_WASM_VERIFY_WITH_ED25519_PUBKEY
+  static const auto ed25519_pubkey = hex2pubkey<32>(PROXY_WASM_VERIFY_WITH_ED25519_PUBKEY);
+  return ed25519_pubkey;
+#else
+  if (!pubkey.empty()) {
+    char pubkey_char[65];
+    strcpy(pubkey_char, pubkey.c_str());
+    return hex2pubkey<32>(pubkey_char);
```

Comment:
`hex2pubkey` is a `constexpr` template written to decode hex literal at the build-time with zero-cost at runtime.

But if you have to copy all the bytes to use it, then it kind of misses the point, and it would be better to add `hexstr2pubkey` function that can decode it in a single pass (basically, a runtime equivalent of `hex2pubkey`).